### DISCUSSION TITLE
ui.tree pre-selected nodes (or programmatically selecting nodes)

### DISCUSSION THREAD

**Discussion by MohAid** [zauberzeug/nicegui#3436] [Ideas / Feature Requests]
I am unsure if it's related to NICEGUI, but I suggest enhancing the documentation with an explanation for the "node object" structure (in tree element).
I needed to select (check) some nodes programmatically and I dived for a good time until I found `tree._props['ticked'].append(id)`
Thanks in advance for your efforts.

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @MohAid!
We should probably introduce a proper API for ticking and selecting nodes programmatically (similar to `expand()` and `collapse()`).
Would anyone like to create a pull request?

**Reply by arunaseva** [reply]
> Would anyone like to create a pull request?

May I try it, but I don't have much experience.

Remarks:

**Selecting node**:

As I see in Quasar manual (https://quasar.dev/vue-components/tree/#selection-vs-ticking-expansion) only one node can be selected at once. I can select or deselect only one node. Selecting new node remove selection of the earlier selected node (if selected). So:

At /workspaces/nicegui/nicegui/elements/tree.py:44
`self._props['selected'] = []`
should be:
`self._props['selected'] = ''`

Suggested methods:
```py
select(node_key: str)
deselect()
```

**Ticking nodes**:

Suggested methods:
```py
tick(node_keys: Set[str])
untick(node_keys: Set[str])
```

But `tick()` behaivor depends on `tick_strategy` and `tick_strategy` value can be `leaf-filtered` and
`leaf-filtered` is only applicable if`filter` or `filter-method` prop is defined.
Should python API contain these two new class properties as well?

**Reply by arunaseva** [reply]
Link to Pull Request: https://github.com/zauberzeug/nicegui/pull/3476

**Reply by falkoschindler** [reply]
Regarding the node filter: I just created PR #3481 to support this feature in Python.

**Comment by arunaseva** [comment]
Remove selection:

`
tree.props(add="selected=None")
`

Add selection:

```python
report_id = "report_id_of_node_to_select"   # <<< Change this
tree.props(add=f"selected={report_id}")
```

**Comment by falkoschindler** [comment]
This feature has just been released in version 1.4.35.
